the-collab-lab/tcl-37-smart-shopping-list
# Soon, Kind of Soon, Not Soon Buttons

## Description

Create Soon, Kind of Soon, Not Soon Button Component(s) for the Add Item page


This PR takes the radio buttons for soon, kind of soon, not soon and gives them the look of a button. I used CSS to hide the input field for the radio button, and styled the label as a box. While the radio itself is not a controlled React component for the purposes of being :checked, it does still pass a value onChange to update the input state in the db call.

I chose to leave the checked attribute controlled by the radio button elements grouped with name="days". Since all the buttons are identified with the name="days", if you click one button it is :checked and styled. In this case the CSS for the :checked radio button is a box shadow corresponding to the label's text color.  This choice kept the code simple, and prevented the overuse of props and state to hold the state for :checked. (It already worked, as is.)

Figma design:
![image](https://user-images.githubusercontent.com/67841237/155779607-14d28432-b7b1-4b73-b793-4e71209609cc.png)

## Related Issue

closes #41 



## Acceptance Criteria

- [X] Radio buttons look like regular buttons
- [X] The buttons function as a group. If one is checked, it is styled as checked.
- [X] If a button is checked, when an item is submitted, the value of soon/not-soon/kind-of-soon that is checked is passed to the item in the list.



## Type of Changes





|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |
|     | :link: Update dependencies |


## Updates

### Before
![Screen Shot 2022-02-26 at 11 13 34 AM](https://user-images.githubusercontent.com/67841237/155850430-2f4b6b1d-68f8-4a38-bf19-022f907aad94.png)



### After
<img width="534" alt="Screen Shot 2022-03-02 at 1 07 11 PM" src="https://user-images.githubusercontent.com/67841237/156421980-35aaf150-31d0-4ea1-9c1c-cd20e0c8613a.png">



## Testing Steps / QA Criteria

1. You will need to pull this down to properly test. The buttons don't operate the same in preview.
2. Add an item to the list. (You should expect to see "Soon" selected as a default).
3. Click the Not Soon or Kind of Soon Buttons and Submit the item. (When clicked, it holds a checked attribute and you should see a colored box shadow.
4. You should expect the item to be on the list (in List View) with the correct amount of days. (In the right soon/not-soon, kind-of-soon category.)
